gov2/testtools/scenarios.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package testtools
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+)
+
+// IScenarioTest defines callback functions that an individual scenario test must
+// implement.
+//
+// SetupDataAndStubs sets up test data specific to the test and returns a list of stubs
+// that specify the expected calls that are made during a scenario run.
+//
+// RunSubTest runs the actual code under test.
+type IScenarioTest interface {
+	SetupDataAndStubs() []Stub
+	RunSubTest(stubber *AwsmStubber)
+}
+
+// RunScenarioTests runs the scenario multiple times. The first time, it
+// runs with no errors. In subsequent runs, it specifies that each stub in the sequence
+// should raise an error and verifies the results.
```

Comment:
should raise an error and verifies the results. -->  should raise an error, and it verifies the results.

Reply:
fixed